lib/collection-methods.js

```diff
@@ -55,7 +55,7 @@ Object.defineProperty(iteratorPrototype, Symbol.iterator, {
 
 ['entries', 'keys', 'values'].forEach(function(methodName) {
     var method = function() {
-        const isSet = this instanceof Realm.Set;
+        const isSet = (this.constructor && this.constructor.name && this.constructor.name === 'Set');
```

Comment:
I would suggest that we add a comment why `instanceof Realm.Set` doesn't work.

Reply:
Wouldn't it be better to repeat the pattern you introduced for `set-methods.js`? https://github.com/realm/realm-js/blob/ca7051f8dc566101a8a89444575d16d5167ed83f/lib/set-methods.js#L21-L30

Basically refactoring `collections-methods.js` to export a function that takes the `Realm` constructor as an argument.